MultiSigWalletOverride/docker/immutability/config/entrypoint.sh

```diff
@@ -1,20 +1,123 @@
-#!/bin/sh
+#!/bin/bash
 
 # Vault running in the container must listen on a different port.
-export VAULT_ADDR="http://127.0.0.1:8900"
 
-nohup vault server -dev -dev-root-token-id="totally-secure" \
-  -log-level=debug -config=/home/vault/config/ -dev-listen-address="0.0.0.0:8900" &
-VAULT_PID=$!
+VAULT_CREDENTIALS="/home/vault/config/unseal.json"
+
+CONFIG_DIR="/home/vault/config"
+
+CA_CERT="$CONFIG_DIR/ca.crt"
+CA_KEY="$CONFIG_DIR/ca.key"
+TLS_KEY="$CONFIG_DIR/my-service.key"
+TLS_CERT="$CONFIG_DIR/my-service.crt"
+CONFIG="$CONFIG_DIR/openssl.cnf"
+CSR="$CONFIG_DIR/my-service.csr"
+
+export VAULT_ADDR="https://127.0.0.1:8200"
+export VAULT_CACERT="$CA_CERT"
+
+function create_config {
+
+	cat > "$CONFIG" << EOF
+
+[req]
+default_bits = 2048
+encrypt_key  = no
+default_md   = sha256
+prompt       = no
+utf8         = yes
+
+# Speify the DN here so we aren't prompted (along with prompt = no above).
+distinguished_name = req_distinguished_name
+
+# Extensions for SAN IP and SAN DNS
+req_extensions = v3_req
+
+# Be sure to update the subject to match your organization.
+[req_distinguished_name]
+C  = TH
+ST = Bangkok
+L  = Vault
+O  = omiseGO
+CN = localhost
+
+# Allow client and server auth. You may want to only allow server auth.
+# Link to SAN names.
+[v3_req]
+basicConstraints     = CA:FALSE
+subjectKeyIdentifier = hash
+keyUsage             = digitalSignature, keyEncipherment
+extendedKeyUsage     = clientAuth, serverAuth
+subjectAltName       = @alt_names
+
+# Alternative names are specified as IP.# and DNS.# for IPs and
+# DNS accordingly.
+[alt_names]
+IP.1  = 127.0.0.1
+IP.2  = 192.168.64.1
+IP.3  = 192.168.122.1
+DNS.1 = localhost
+EOF
+}
+
+function gencerts {
 
-count=1
-while [ "$count" -le 60 ]; do
-	if vault status; then break; fi
-	count=$((count+1))
-	sleep 0.5
-done
+    create_config
+	openssl req \
+	-new \
+	-sha256 \
+	-newkey rsa:2048 \
+	-days 120 \
+	-nodes \
+	-x509 \
+	-subj "/C=US/ST=Maryland/L=Vault/O=My Company CA" \
+	-keyout "$CA_KEY" \
+	-out "$CA_CERT"
 
-vault status
+	# Generate the private key for the service. Again, you may want to increase
+	# the bits to 2048.
+	openssl genrsa -out "$TLS_KEY" 2048
+
+	# Generate a CSR using the configuration and the key just generated. We will
+	# give this CSR to our CA to sign.
+	openssl req \
+	-new -key "$TLS_KEY" \
+	-out "$CSR" \
+	-config "$CONFIG"
+
+	# Sign the CSR with our CA. This will generate a new certificate that is signed
+	# by our CA.
+	openssl x509 \
+	-req \
+	-days 120 \
+	-in "$CSR" \
+	-CA "$CA_CERT" \
+	-CAkey "$CA_KEY" \
+	-CAcreateserial \
+	-sha256 \
+	-extensions v3_req \
+	-extfile "$CONFIG" \
+	-out "$TLS_CERT"
+
+	openssl x509 -in "$TLS_CERT" -noout -text
+
+	# rm openssl.cnf
+
+#  chown -R nobody:nobody $CONFIG_DIR && chmod -R 777 $CONFIG_DIR
+}
+
+gencerts
```

Comment:
I am not a script expert so I don't know what should be best practice, but shall we run commands after all function declaration so we don't mix reading command and function declaration together ??

Reply:
this is really just a copy/clone of *what comes with vault*. and it's purpose is to be able to standup and test the "migration integration" really works!